esdt-safe/src/lib.rs

```diff
@@ -50,45 +32,43 @@ pub trait EsdtSafe {
     }
 
     #[endpoint(addTokenToWhitelist)]
-    fn add_token_to_whitelist(&self, token_identifier: TokenIdentifier) -> SCResult<()> {
+    fn add_token_to_whitelist(&self, token_id: TokenIdentifier) -> SCResult<()> {
         only_owner!(self, "only owner may call this function");
 
-        self.token_whitelist().insert(token_identifier);
+        self.token_whitelist().insert(token_id);
 
         Ok(())
     }
 
     #[endpoint(removeTokenFromWhitelist)]
-    fn remove_token_from_whitelist(&self, token_identifier: TokenIdentifier) -> SCResult<()> {
+    fn remove_token_from_whitelist(&self, token_id: TokenIdentifier) -> SCResult<()> {
         only_owner!(self, "only owner may call this function");
 
-        self.token_whitelist().remove(&token_identifier);
+        self.token_whitelist().remove(&token_id);
 
         Ok(())
     }
 
     #[endpoint(getNextPendingTransaction)]
-    fn get_next_pending_transaction(&self) -> SCResult<OptionalResult<Transaction<BigUint>>> {
+    fn get_next_pending_transaction(&self) -> SCResult<Option<Transaction<BigUint>>> {
```

Comment:
What's wrong with the OptionalResult?

Reply:
Unfortunately, OptionalResult can't be used in SC proxies at the moment, so I'm switching to Option until it's resolved. (I'm calling this endpoint from the multisig SC through a proxy)